File: Data Structures & Algorithms/palindrome-permutation/submission-1.py
class Solution:
    def canPermutePalindrome(self, s: str) -> bool:
        mp = {}
        for ch in s:
            mp[ch] = mp.get(ch, 0) + 1
        count = 0
        for index, value in mp.items():
            if value % 2 != 0:
                count += 1
        if count > 1:
            return False
        return True

File: Data Structures & Algorithms/validate-parentheses/submission-1.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<int> st_data;
        unordered_map <char, char> unmap = {
            { '}', '{' },
            { ']', '[' },
            { ')', '(' },
        };

        for(char c : s ) {
            if(unmap.count(c)) {
                if (!st_data.empty() && st_data.top() == unmap[c]) {
                    st_data.pop();
                } else {
                    return false;
                }
            } else {
                st_data.push(c);
            }
        }

        return st_data.empty();
    }
};
